Daniel db functionality

nolannish opened the pull request:
Added functionality to teacherRegistration, studentRegistration, and login pages. We are now able to enter information into custom login page, and then have that information passed to the database, as well as pulling information from the database for login authentication. This can be further optimized in the future, but having the working product right now is a big step.

Daniel-Penner committed:
Merge pull request #102 from chpsmstr/main

pull from main

Daniel-Penner committed:
make teacher registration send

nolannish committed:
Custom register pages inserts data into database

Register pages are now at a point in which you can enter information into the fields, and then on clicking submit button, page takes you to the proper dashboard you are supposed to redirect do, but also form data present in register fields are inserted into the database. Also removed route.ts pages from StudentRegistration and TeacherRegistration, as they become obsolete through the use of server actions that are now handling database interaction.

Daniel-Penner committed:
Merge pull request #105 from chpsmstr/main

bringing the new main functionality over

nolannish committed:
Added functionality to login page

Custom login page works now, with routing to specific dashboard depending on information entered i.e., if information entered in login comes from student, then redirects to teacher dashboard. There can be further optimization done to this, but as of right now having it be functional is good.

Daniel-Penner reviewed (approved):
Awesome that registration is now working!

Daniel-Penner committed:
Added database functionality to Admin class registration page

Admin can now see a list of students pulled from the database, file has been heavily refactored.
